### PR TITLE
fix: convergence 

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/__tests__/fcnnls_deconv.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { X as X2, Y as Y2 } from './deconv-examples/data_example2';
+import { X as X3, Y as Y3 } from './deconv-examples/data_example3';
+
+describe('fcnnls deconv-examples run', () => {
+  it('example 1', () => {
+    const result = fcnnls(X1, Y1);
+    expect(result).toBeDefined();
+  });
+  it('example 2', () => {
+    const result = fcnnls(X2, Y2, { gradientTolerance: 1e-4 });
```
Comment (lpatiny) [inline src/__tests__/fcnnls_deconv.test.js:13]: Does it crash with default gradientTolerance ?

Comment (ghost) [reply src/__tests__/fcnnls_deconv.test.js:13]: Yes, both example 2 and 3. 

We can make it `1e-4`, but this value will depend in the specific example.

File: src/__tests__/deconv-examples/data_example1.js [hunk 2]
```diff
@@ -0,0 +1,1133 @@
+export const X = [
```
Comment (lpatiny) [inline src/__tests__/deconv-examples/data_example1.js:1]: Could you add a test with all the values (in X and Y) - 1. Adding 1 was a way to circumvent previous bug.

**Comment by lpatiny** [comment]
Does the paper speaks about gradientTolerance ?

Is this value (gradientTolerance) depending of the amplitude of the input X and Y ? Meaning that if all the values are between 0 and 1 or between 0 and 1e9 should it be the same gradientTolerance) ?

**Pull request merged**